internal/utils/utils.go

```diff
@@ -529,7 +534,7 @@ func DownloadFile(url string) (string, error) {
 	tempParentDir := filepath.Join(os.TempDir(), uuid.NewString())
 	internalio.MkdirAll(tempParentDir)
 
-	tempFile, err := os.Create(filepath.Join(tempParentDir, path.Base(url)))
+	tempFile, err := os.Create(filepath.Join(tempParentDir, uuid.NewString()))
```

Comment:
This was failing on windows when the artifact was downloaded from a GCP bucket, as the URL contained URL params, escaped chars, etc...